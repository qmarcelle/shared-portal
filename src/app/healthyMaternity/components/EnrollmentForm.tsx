import { Button } from '@/components/foundation/Button';
import { Card } from '@/components/foundation/Card';
import { Checkbox } from '@/components/foundation/Checkbox';
import { Column } from '@/components/foundation/Column';
import { Divider } from '@/components/foundation/Divider';
import { Header } from '@/components/foundation/Header';
import { Radio } from '@/components/foundation/Radio';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import { TextField } from '@/components/foundation/TextField';
import { IComponent } from '@/components/IComponent';
import Image from 'next/image';
import { useState } from 'react';
import { successIcon } from '../../../components/foundation/Icons';

interface EnrollmentFormProps extends IComponent {
  accessCode: string;
}

export const EnrollmentForm = ({ accessCode }: EnrollmentFormProps) => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [myBool, setmyBool] = useState(false);

  function navigateContent() {
    setmyBool(!myBool);
  }

  const [myBool1, setmyBool1] = useState(false);

  function navigateSuccessPage() {
    setmyBool1(!myBool1);
  }

  const [isChecked, setIsChecked] = useState(false);

  const checkHandler = () => {
    setIsChecked(!isChecked);
  };

  function memberInfo() {
    return (
      <Card className="large-section">
        <section className="w-4/5 m-auto" id="stepSection">
          <TextBox text="1" className="stepRoundBlue" />
          <Divider size={5} className="stepLineGrey" />
          <TextBox text="2" className="stepRoundGrey" />
          <Divider size={5} className="stepLineGrey" />
          <TextBox text="3" className="stepRoundGrey" />
          <Spacer size={18} />
          <TextBox text="Member Information" />
          <Spacer size={18} />
          <TextField label="Member Name" />
          <Spacer size={8} />
          <TextField label="Date of Birth (MM/DD/YYYY)" />
          <Spacer size={8} />
          <TextField label="Due Date of Baby (MM/DD/YYYY)" />
          <Spacer size={8} />
          <TextField label="Email Address" />
          <Spacer size={8} />
          <TextField label="Phone Number" />
          <Checkbox label="By checking this box I agree to BlueCross, its affiliates and its service providers sending me communications via email. Unencrypted email may possibly be intercepted and read by people other than those it's addressed to." />
          <Spacer size={8} />
          <Button label="Next" callback={navigateContent} />
        </section>
      </Card>
    );
  }

  function contactMethod() {
    return (
      <Card className="large-section">
        <section className="w-4/5 m-auto" id="stepSection">
          <TextBox text="1" className="stepRoundBlue" />
          <Divider size={5} className="stepLineBlue" />
          <TextBox text="2" className="stepRoundBlue" />
          <Divider size={5} className="stepLineGrey" />
          <TextBox text="3" className="stepRoundGrey" />
          <Spacer size={18} />
          <TextBox text="Preferred Contact Method" />
          <Spacer size={18} />
          <TextBox text="I prefer that my healthy maternity nurse contacts me via:" />
          <Spacer size={8} />
          <Radio
            selected={false}
            callback={checkHandler}
            label="Messages in the CareTN app"
            subLabel="You'll need a compatible Apple or Android mobile device. Your nurse may call you but most of the time they'll message you through the app."
          />
          <Spacer size={12} />
          <Radio
            label="Phone call"
            selected={false}
            subLabel="Your nurse will call you during your pregnancy."
          />
          <Spacer size={18} />
          {isChecked && (
            <>
              <TextBox text="Check the box to get help downloading the CareTN app:" />
              <Checkbox label="Text me a link to the CareTN app." />
            </>
          )}
          <Spacer size={18} />
          <Row>
            <Button label="Back" type="secondary" callback={navigateContent} />
            <Spacer size={18} axis="horizontal" />
            <Button label="Next" callback={navigateSuccessPage} />
          </Row>
        </section>
      </Card>
    );
  }

  function successContent() {
    return (
      <Card className="large-section">
        <section className="w-4/5 m-auto text-center" id="stepSection">
          <TextBox text="1" className="stepRoundBlue" />
          <Divider size={5} className="stepLineBlue" />
          <TextBox text="2" className="stepRoundBlue" />
          <Divider size={5} className="stepLineBlue" />
          <TextBox text="3" className="stepRoundBlue" />
          <Spacer size={18} />
          <Column className="items-center">
            <Image className="size-[80px]" src={successIcon} alt="success" />
            <Spacer size={24} />
            <Header
              className="title-2"
              text="You're Enrolled in the Healthy Maternity Program"
            />
            <Spacer size={16} />
            <Column>
              <TextBox
                type="body-2"
                text="We'll review your information and a nurse will reach out to you soon."
              />
              <Spacer size={18} />
              <Divider />
              <Spacer size={18} />
              <TextBox text="Next Steps" />
              <Spacer size={8} />
              <TextBox
                type="body-2"
                text="Download the CareTN app and enter the access code below to set up your account."
              />
              <Spacer size={9} />
              <Column className="py-3 px-7 accessCodeBackground">
                <TextBox text="Access Code" type="body-2" />
                <TextBox text={accessCode} className="font-bold" />
              </Column>
            </Column>
          </Column>
        </section>
      </Card>
    );
  }

  return (
    <Card className="large-section">
      <section>
        <Header
          text="Enroll in Health Maternity"
          className="title-2 font-bold"
        />
        <Spacer size={8} />
        <TextBox text="Please confirm your details below." />
        <Spacer size={16} />
        {!myBool && memberInfo()}
        {myBool && !myBool1 && contactMethod()}
        {myBool1 && successContent()}
      </section>
    </Card>
  );
};

export default EnrollmentForm;
