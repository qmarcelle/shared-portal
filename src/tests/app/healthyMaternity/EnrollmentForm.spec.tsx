/* eslint-disable quotes */
import EnrollmentForm from '@/app/(protected)/(common)/member/healthyMaternity/components/EnrollmentForm';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';

const renderUI = () => {
  return render(<EnrollmentForm accessCode="BlueAccess" />);
};

describe('EnrollmentForm Section', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    expect(screen.getByText('Member Information')).toBeVisible();
    expect(
      screen.getByText('Please confirm your details below.'),
    ).toBeVisible();
    expect(
      screen.getByText(
        "By checking this box I agree to BlueCross, its affiliates and its service providers sending me communications via email. Unencrypted email may possibly be intercepted and read by people other than those it's addressed to.",
      ),
    );
    fireEvent.click(screen.getByRole('button', { name: /Next/i }));
    expect(
      screen.getByText(
        "You'll need a compatible Apple or Android mobile device. Your nurse may call you but most of the time they'll message you through the app.",
      ),
    ).toBeVisible();
    expect(screen.getByText('Preferred Contact Method')).toBeVisible();
    expect(
      screen.getByText(
        'I prefer that my healthy maternity nurse contacts me via:',
      ),
    ).toBeVisible();
    expect(screen.getByText('Phone call')).toBeVisible();
    expect(
      screen.getByText('Your nurse will call you during your pregnancy.'),
    ).toBeVisible();
    fireEvent.click(screen.getByRole('button', { name: /Next/i }));
    expect(
      screen.getByText("You're Enrolled in the Healthy Maternity Program"),
    ).toBeVisible();
    expect(
      screen.getByText(
        "We'll review your information and a nurse will reach out to you soon.",
      ),
    ).toBeVisible();
    expect(component).toMatchSnapshot();
  });
});
