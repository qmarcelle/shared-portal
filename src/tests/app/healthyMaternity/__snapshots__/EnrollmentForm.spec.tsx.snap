// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnrollmentForm Section should render the UI correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-main large-section"
      >
        <section>
          <h1
            class="title-1 title-2 font-bold"
          >
            Enroll in Health Maternity
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <p
            class="body-1"
            style="display: block; max-width: 100%;"
          >
            Please confirm your details below.
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <div
            class="card-main large-section"
          >
            <section
              class="w-4/5 m-auto text-center stepSection"
            >
              <p
                class="body-1 stepRoundBlue"
                style="display: block; max-width: 100%;"
              >
                1
              </p>
              <div
                class="stepLineBlue"
                style="width: 100%; height: 5px; background-color: rgb(204, 204, 204);"
              />
              <p
                class="body-1 stepRoundBlue"
                style="display: block; max-width: 100%;"
              >
                2
              </p>
              <div
                class="stepLineBlue"
                style="width: 100%; height: 5px; background-color: rgb(204, 204, 204);"
              />
              <p
                class="body-1 stepRoundBlue"
                style="display: block; max-width: 100%;"
              >
                3
              </p>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
              />
              <div
                class="flex flex-col items-center"
              >
                <img
                  alt=""
                  class="size-[80px]"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
                />
                <h1
                  class="title-1 title-2"
                >
                  You're Enrolled in the Healthy Maternity Program
                </h1>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-col"
                >
                  <p
                    class="body-2"
                    style="display: block; max-width: 100%;"
                  >
                    We'll review your information and a nurse will reach out to you soon.
                  </p>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
                  />
                  <div
                    class=""
                    style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                  />
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
                  />
                  <p
                    class="body-1"
                    style="display: block; max-width: 100%;"
                  >
                    Next Steps
                  </p>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                  />
                  <p
                    class="body-2"
                    style="display: block; max-width: 100%;"
                  >
                    Download the CareTN app and enter the access code below to set up your account.
                  </p>
                  <span
                    style="display: block; width: 1px; min-width: 1px; height: 9px; min-height: 9px;"
                  />
                  <div
                    class="flex flex-col py-3 px-7 accessCodeBackground"
                  >
                    <p
                      class="body-2"
                      style="display: block; max-width: 100%;"
                    >
                      Access Code
                    </p>
                    <p
                      class="body-1 font-bold"
                      style="display: block; max-width: 100%;"
                    >
                      BlueAccess
                    </p>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-main large-section"
    >
      <section>
        <h1
          class="title-1 title-2 font-bold"
        >
          Enroll in Health Maternity
        </h1>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
        />
        <p
          class="body-1"
          style="display: block; max-width: 100%;"
        >
          Please confirm your details below.
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <div
          class="card-main large-section"
        >
          <section
            class="w-4/5 m-auto text-center stepSection"
          >
            <p
              class="body-1 stepRoundBlue"
              style="display: block; max-width: 100%;"
            >
              1
            </p>
            <div
              class="stepLineBlue"
              style="width: 100%; height: 5px; background-color: rgb(204, 204, 204);"
            />
            <p
              class="body-1 stepRoundBlue"
              style="display: block; max-width: 100%;"
            >
              2
            </p>
            <div
              class="stepLineBlue"
              style="width: 100%; height: 5px; background-color: rgb(204, 204, 204);"
            />
            <p
              class="body-1 stepRoundBlue"
              style="display: block; max-width: 100%;"
            >
              3
            </p>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
            />
            <div
              class="flex flex-col items-center"
            >
              <img
                alt=""
                class="size-[80px]"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <span
                style="display: block; width: 1px; min-width: 1px; height: 24px; min-height: 24px;"
              />
              <h1
                class="title-1 title-2"
              >
                You're Enrolled in the Healthy Maternity Program
              </h1>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
              />
              <div
                class="flex flex-col"
              >
                <p
                  class="body-2"
                  style="display: block; max-width: 100%;"
                >
                  We'll review your information and a nurse will reach out to you soon.
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
                />
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
                />
                <p
                  class="body-1"
                  style="display: block; max-width: 100%;"
                >
                  Next Steps
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
                />
                <p
                  class="body-2"
                  style="display: block; max-width: 100%;"
                >
                  Download the CareTN app and enter the access code below to set up your account.
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 9px; min-height: 9px;"
                />
                <div
                  class="flex flex-col py-3 px-7 accessCodeBackground"
                >
                  <p
                    class="body-2"
                    style="display: block; max-width: 100%;"
                  >
                    Access Code
                  </p>
                  <p
                    class="body-1 font-bold"
                    style="display: block; max-width: 100%;"
                  >
                    BlueAccess
                  </p>
                </div>
              </div>
            </div>
          </section>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
